x/staking/keeper/historical_info.go

```diff
@@ -73,9 +38,9 @@ func (k Keeper) IterateHistoricalInfo(ctx context.Context, cb func(types.Histori
 // GetAllHistoricalInfo returns all stored HistoricalInfo objects.
 func (k Keeper) GetAllHistoricalInfo(ctx context.Context) ([]types.HistoricalInfo, error) {
 	var infos []types.HistoricalInfo
-	err := k.IterateHistoricalInfo(ctx, func(histInfo types.HistoricalInfo) bool {
-		infos = append(infos, histInfo)
-		return false
+	err := k.HistoricalInfo.Walk(ctx, nil, func(key int64, info types.HistoricalInfo) (stop bool, err error) {
```

Comment:
We should check for collections Invalid Iter error. the SDK did not error on empty results, collection does.